src/types/resolveStatements.ts

```diff
@@ -180,6 +180,11 @@ function processStatements(
                 );
             }
 
+            const staticFunction = getAllStaticFunctions(ctx);
+            if (Object.values(staticFunction).some((f) => f.name === s.name)) {
+                throwError(`Variable shadows existing function: ${s.name}`, s.ref);
+            
+            }
```

Comment:
This looks a bit like an ad hoc approach to me. For instance, would it work for contract functions? This should be solved with a unified variable context if you ask me. So, a bit of refactoring is needed here